File: DP1/11_Searching/binarySearch.js
function binarySearch(arr, n) {
    let start = 0;
    let end = arr.length;
    let found = false;

    while(!found && start <= end) {
        let mid = parseInt((start + end) / 2);

        if(arr[mid] === n) {
            found = true;
        } else if(arr[mid] < n) {
            start = mid + 1;
        } else {
            end = mid - 1;
        }
    }

    return found;
}

function main() {
    const arr = [1,2,3,4,5,6,7]
    const n = 3;

    console.log(binarySearch(arr, n));
}


main();

File: DP1/11_Searching/binarySearch.go
package main;

import "fmt";

func binarySearch(arr []int, n int) bool {
  found := false
  start := 0
  end := len(arr)

  for !found && start <= end {
    mid := (start + end) / 2

    if arr[mid] == n {
      found = true
    } else if arr[mid] < n {
      start = mid + 1
    } else {
      end = mid - 1
    }
  }

  return found
}

func main() {
  var array = []int{1,3,5}
  n := 1
  fmt.Println(binarySearch(array, n))
}
